components/pages/explorer/BamTable.tsx

```diff
@@ -99,19 +105,25 @@ const ToggleButtonPanel = ({
 	</div>
 );
 
-const BamTable = () => {
+const BamTable = ({ file }: { file: FileTableData | null }) => {
 	const theme = useTheme();
-
+	const [fileMetaData, setFileMetaData] = useState<FileMetaData | null>(null);
 	const [elementState, toggleElementState] = useState(initElementState);
-	const [loading, setLoading] = useState(false);
+	const [loading, setLoading] = useState(true);
 
-	// TODO: This will be replaced by File data found in Arranger and passed down through context / parent components
-	const fileUrl = 'https://s3.amazonaws.com/iobio/NA12878/NA12878.autsome.bam';
-	const fileName = fileUrl.split('/').pop();
+	const fileUrl = fileMetaData?.parts[0]?.url || null;
 
-	useEffect(() => {
-		setLoading(false);
-	}, []);
+	// Todo: Update fileName definition
+	const fileName = file?.id || fileUrl?.split('/').pop()?.split('?')[0];
+
+	const loadAndSetFile = async (file: FileTableData) => {
+		await getFileMetaData(file).then((data) => {
+			if (data) {
+				setFileMetaData(data);
+				setLoading(false);
+			}
+		});
+	};
```

Comment:
mixing of styles - Promise vs await?
`await` and `then` are two distinct means to an end

Reply:
indeed, the await here doesn't seem to serve any purpose, for the awaited value is not used anywhere. Does this suggestion work the same? if so, there may be a better way to implement this whole segment, using React lifecycle methods.

```suggestion
	const loadAndSetFile = (file: FileTableData) => new Promise((resolve, reject) => {
		getFileMetaData(file).then((data) => {
			if (data) {
				setFileMetaData(data);
				setLoading(false);
				return resolve();
			}
			reject();
		});
	});
```

Reply:
[Straighten out promise/await mix](https://github.com/overture-stack/stage/pull/210/commits/39050139ab248ae218887c4bfa5f452e76909768)[Straighten out promise/await mix](https://github.com/overture-stack/stage/pull/210/commits/39050139ab248ae218887c4bfa5f452e76909768)